tests/stub/summary/test_summary.py

```diff
@@ -247,6 +235,793 @@ def test_multiple_notifications(self):
+        notification = self.make_test_notification(severity=severity)
+        del notification[del_key]
+        in_notifications = [notification]
+        summary = self._get_summary(
+            "summary_with_notifications.script",
+            vars_={
+                "#NOTIFICATIONS#": json.dumps(in_notifications)
+            }
+        )
+
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            if severity == "WARNING":
+                self.assert_is_success(summary.gql_status_objects[1])
+                return_index = 0
+            else:
+                self.assert_is_success(summary.gql_status_objects[0])
+                return_index = 1
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+            return_index = 1
+        return summary.gql_status_objects[return_index]
+
+    def test_notification_with_missing_severity(self):
+        status = self._test_notification_with_missing_data(
+            "severity", severity="INFORMATION"
+        )
+        self.assert_is_test_notification_as_gql_status_object(
+            status, severity=None, parsed_severity="UNKNOWN"
+        )
+
+    def test_notification_with_missing_category(self):
+        status = self._test_notification_with_missing_data("category")
+        self.assert_is_test_notification_as_gql_status_object(
+            status, category=None, parsed_category="UNKNOWN"
+        )
+
+    def test_notification_with_missing_position(self):
+        status = self._test_notification_with_missing_data("position")
+        self.assert_is_test_notification_as_gql_status_object(
+            status, raw_pos=None
+        )
+
+    def test_warn_with_missing_description(self):
+        status = self._test_notification_with_missing_data("description")
+        self.assert_is_test_notification_as_gql_status_object(
+            status, description="warn: warning - unknown warning",
+        )
+
+    def test_info_with_missing_description(self):
+        status = self._test_notification_with_missing_data(
+            "description", severity="INFORMATION"
+        )
+        self.assert_is_test_notification_as_gql_status_object(
+            status, description="info: informational - unknown notification",
+            severity="INFORMATION", parsed_severity="INFORMATION",
+        )
+
+    def test_info_fallback_with_missing_description(self):
+        status = self._test_notification_with_missing_data(
+            "description", severity="BANANA"
+        )
+        self.assert_is_test_notification_as_gql_status_object(
+            status, description="info: informational - unknown notification",
+            severity="BANANA", parsed_severity="UNKNOWN",
```

Comment:
```suggestion
            severity="BANANA", parsed_severity="None",
```
Java driver does not have the `UNKNOWN` parsed value, it returns an empty `Optional` is such cases. The backend can simulate `UNKNOWN` value for Testkit, but it lead to some other tests failure.